diff --git a/nheqminer/main.cpp b/nheqminer/main.cpp
index 6f7a230..7d10ff4 100644
--- a/nheqminer/main.cpp
+++ b/nheqminer/main.cpp
@@ -37,7 +37,7 @@ namespace src = boost::log::sources;
 namespace attrs = boost::log::attributes;
 namespace keywords = boost::log::keywords;
 
-#ifdef __linux__
+#if (defined(__linux__) || defined(__APPLE__))
 #define __cpuid(out, infoType)\
 	asm("cpuid": "=a" (out[0]), "=b" (out[1]), "=c" (out[2]), "=d" (out[3]): "a" (infoType));
 #define __cpuidex(out, infoType, ecx)\
