diff --git a/nheqminer/crypto/common.h b/nheqminer/crypto/common.h
index 4b2b16a..70dfdba 100644
--- a/nheqminer/crypto/common.h
+++ b/nheqminer/crypto/common.h
@@ -13,7 +13,7 @@
 #include <assert.h>
 
 //#include "sodium.h"
-#ifdef WIN32
+#if (defined(WIN32) || defined(__APPLE__))
 #include "compat/endian.h"
 #else
 #include <endian.h>
diff --git a/nheqminer/main.cpp b/nheqminer/main.cpp
index 6f7a230..7d10ff4 100644
--- a/nheqminer/main.cpp
+++ b/nheqminer/main.cpp
@@ -37,7 +37,7 @@ namespace src = boost::log::sources;
 namespace attrs = boost::log::attributes;
 namespace keywords = boost::log::keywords;
 
-#ifdef __linux__
+#if (defined(__linux__) || defined(__APPLE__))
 #define __cpuid(out, infoType)\
 	asm("cpuid": "=a" (out[0]), "=b" (out[1]), "=c" (out[2]), "=d" (out[3]): "a" (infoType));
 #define __cpuidex(out, infoType, ecx)\
diff --git a/nheqminer/serialize.h b/nheqminer/serialize.h
index ecb6cf1..77ccae4 100644
--- a/nheqminer/serialize.h
+++ b/nheqminer/serialize.h
@@ -6,7 +6,7 @@
 #ifndef BITCOIN_SERIALIZE_H
 #define BITCOIN_SERIALIZE_H
 
-#ifdef WIN32
+#if (defined(WIN32) || defined(__APPLE__))
 #include "compat/endian.h"
 #else
 #include <endian.h>
