diff --git a/CMakeLists.txt b/CMakeLists.txt
index f906675..1bcd76b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,9 +184,17 @@ if (USE_CPU_TROMP)
 endif()
 if (USE_CPU_XENONCAT)
     add_library ( xenoncat_avx1 SHARED IMPORTED GLOBAL )
-    set_target_properties ( xenoncat_avx1 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_linux/equihash_avx1.o" )
+    if (APPLE)
+        set_target_properties ( xenoncat_avx1 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_mac/equihash_avx1.o" )
+    else()
+        set_target_properties ( xenoncat_avx1 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_linux/equihash_avx1.o" )
+    endif(APPLE)
     add_library ( xenoncat_avx2 SHARED IMPORTED GLOBAL )
-    set_target_properties ( xenoncat_avx2 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_linux/equihash_avx2.o" )
+    if (APPLE)
+        set_target_properties ( xenoncat_avx2 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_mac/equihash_avx2.o" )
+    else()
+        set_target_properties ( xenoncat_avx2 PROPERTIES IMPORTED_LOCATION "../nheqminer/cpu_xenoncat/asm_linux/equihash_avx2.o" )
+    endif(APPLE)
     target_link_libraries(${PROJECT_NAME} cpu_xenoncat xenoncat_avx1 xenoncat_avx2)
 endif()
 if (USE_CUDA_TROMP)
diff --git a/cpu_xenoncat/asm_mac/assemble.sh b/cpu_xenoncat/asm_mac/assemble.sh
new file mode 100644
index 0000000..35fe998
--- /dev/null
+++ b/cpu_xenoncat/asm_mac/assemble.sh
@@ -0,0 +1,4 @@
+fasm -m 1280000 ../asm_linux/equihash_avx1.asm equihash_avx1.elf.o
+fasm -m 1280000 ../asm_linux/equihash_avx2.asm equihash_avx2.elf.o
+objconv -fmacho64 -nu equihash_avx1.elf.o equihash_avx1.o
+objconv -fmacho64 -nu equihash_avx2.elf.o equihash_avx2.o
