diff --git a/nheqminer/crypto/common.h b/nheqminer/crypto/common.h
index 4b2b16a..70dfdba 100644
--- a/nheqminer/crypto/common.h
+++ b/nheqminer/crypto/common.h
@@ -13,7 +13,7 @@
 #include <assert.h>
 
 //#include "sodium.h"
-#ifdef WIN32
+#if (defined(WIN32) || defined(__APPLE__))
 #include "compat/endian.h"
 #else
 #include <endian.h>
diff --git a/nheqminer/serialize.h b/nheqminer/serialize.h
index ecb6cf1..77ccae4 100644
--- a/nheqminer/serialize.h
+++ b/nheqminer/serialize.h
@@ -6,7 +6,7 @@
 #ifndef BITCOIN_SERIALIZE_H
 #define BITCOIN_SERIALIZE_H
 
-#ifdef WIN32
+#if (defined(WIN32) || defined(__APPLE__))
 #include "compat/endian.h"
 #else
 #include <endian.h>
