diff --git a/CMakeLists.txt b/CMakeLists.txt
index f906675..b63f312 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-project(nheqminer)
+project(nheqminer-macos)
 cmake_minimum_required(VERSION 3.5)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11") #  -Wall
@@ -48,7 +48,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
 endif()
 
 # Common
-include_directories(${nheqminer_SOURCE_DIR}/nheqminer)
+include_directories(${nheqminer-macos_SOURCE_DIR}/nheqminer)
 
 # BOOST
 #find_package(Threads REQUIRED COMPONENTS)
