diff --git a/CMakeLists.txt b/CMakeLists.txt
index f906675..542edf6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,16 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     add_definitions(-O2)
 endif()
 
+# macOS
+if(CMAKE_COMPILER_IS_CLANG)
+    if(APPLE)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native -msse2 -fvisibility=hidden")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -msse2 -fvisibility=hidden")
+        # optimizations
+        add_definitions(-O2)
+    endif(APPLE)
+endif(CMAKE_COMPILER_IS_CLANG)
+
 # Common
 include_directories(${nheqminer_SOURCE_DIR}/nheqminer)
 
