diff --git a/CMakeLists.txt b/CMakeLists.txt
index f906675..39a5aa9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,14 @@ include_directories(${nheqminer_SOURCE_DIR}/nheqminer)
 set(Boost_USE_STATIC_LIBS ON)
 set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
 #set(BUILD_SHARED_LIBRARIES OFF)
+if(CMAKE_COMPILER_IS_GNUCXX)
+if(APPLE)
+set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++")
+else()
 #set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++ -static")
+endif(APPLE)
+endif(CMAKE_COMPILER_IS_GNUCXX)
+
 find_package(Boost REQUIRED COMPONENTS system log_setup log date_time filesystem thread)
 
 if (Boost_FOUND)
